src/LAMMPS.jl

```diff
@@ -75,6 +79,10 @@ const STYLE_GLOBAL = API.LMP_STYLE_GLOBAL
 const STYLE_ATOM = API.LMP_STYLE_ATOM
 const STYLE_LOCAL = API.LMP_STYLE_LOCAL
 
+const BIGINT = API.lammps_extract_setting(C_NULL, "bigint") == 4 ? Int32 : Int64
+const TAGINT = API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64
+const IMAGEINT = API.lammps_extract_setting(C_NULL, "imageint") == 4 ? Int32 : Int64
```

Comment:
When/How can these change? I.e. could I load a different build of lammps where these are now different?

Or are the simply if my platform is 32bit or 64bit? If so we could just use the `Int` alias

Reply:
So it can change when the LAMMPS library changes. I don't recall if `liblammps_path` is a compile-time constant and redefining invalidates the LAMMPS.jl cache.

Reply:
We could in the `__init__` function check that the assumption hasn't changed. That would suffice for now.

Reply:
Changing the Library path requires a julia restart. At least according to our documentation.

Reply:
I'm not really sure how constant a `const` really ist though. Is it redefined on every julia restart or is it possible for it to persist?

Reply:
It will be persisted into the cachefile that Julia emits during precompilation.

Reply:
Does precompilation in Julia include the linking step?

Since liblammp_path is the path of a shared object file, this would have to be linked.
Anecdotally, whenever I've set the library, it's always (as far as I can remember) run precompilation when I've reopened Julia.

Reply:
> We could in the __init__ function check that the assumption hasn't changed. That would suffice for now.

Would we then just error in the `__init__()` function if the integer sizes have changed?
Would this then cause cache invalidation i.e. would a restart of julia then trigger recompilation?
Or would the user have to manually invalidate the LAMMPS.jl cache somehow?

```julia
function __init__()
    BIGINT != (API.lammps_extract_setting(C_NULL, "bigint") == 4 ? Int32 : Int64) && 
         error("The size of the LAMMPS integer type BIGINT has changed! To fix this, restart julia.")
    TAGINT != (API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64) &&
        error("The size of the LAMMPS integer type TAGINT has changed! To fix this, restart julia.")
    IMAGEINT != (API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64) &&
        error("The size of the LAMMPS integer type IMAGEINT has changed! To fix this, restart julia.")
end
````

Reply:
> Or would the user have to manually invalidate the LAMMPS.jl cache somehow?

Yeah the user would need to manually invalidate the cache, but at least we wouldn't get mysterious crashes

Reply:
> Does precompilation in Julia include the linking step?

Julia uses `dlopen` and not static linking

Reply:
@jmeziere 

```suggestion
const BIGINT = API.lammps_extract_setting(C_NULL, "bigint") == 4 ? Int32 : Int64
const TAGINT = API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64
const IMAGEINT = API.lammps_extract_setting(C_NULL, "imageint") == 4 ? Int32 : Int64

function __init__()
    BIGINT != (API.lammps_extract_setting(C_NULL, "bigint") == 4 ? Int32 : Int64) &&
        error("The size of the LAMMPS integer type BIGINT has changed! To fix this, you need to manually invalidate the LAMMPS.jl cache.")
    TAGINT != (API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64) &&
         error("The size of the LAMMPS integer type TAGINT has changed! To fix this, you need to manually invalidate the LAMMPS.jl cache.")
     IMAGEINT != (API.lammps_extract_setting(C_NULL, "tagint") == 4 ? Int32 : Int64) &&
         error("The size of the LAMMPS integer type IMAGEINT has changed! To fix this, you need to manually invalidate the LAMMPS.jl cache.")
end
```

Reply:
I just pushed it up, so it should be good to go. Thanks for your patience with me. This is my first time going through the process.